shared/actions/wallets.js

```diff
@@ -14,13 +14,14 @@ import {getPath} from '../route-tree'
 import {walletsTab} from '../constants/tabs'
 import flags from '../util/feature-flags'
 import {getEngine} from '../engine'
+import {anyWaiting} from '../constants/waiting'
 
 const buildPayment = (state: TypedState, action: any) =>
   RPCTypes.localBuildPaymentLocalRpcPromise({
     amount: state.wallets.buildingPayment.amount,
     // FIXME: Assumes XLM.
-    fromPrimaryAccount: !state.wallets.buildingPayment.from,
-    from: state.wallets.buildingPayment.from,
+    fromPrimaryAccount: state.wallets.buildingPayment.from === Types.noAccountID,
```

Comment:
this was my change, based on your suggestion, but should it now be `state.wallets.builtPayment`?

Reply:
`builtPayment` will always be empty when the send form is first opened, though. the idea is to pass the input from the form to `BuildPaymentLocal` and then use the resolved fields in `SendPaymentLocal`. In this case in `BuildPaymentLocal` the form says `fromPrimaryAccount: true`, we get the address which we put into `builtPayment.from`, which we then send off to `SendPaymentLocal`